### PR TITLE
works 

### PR DESCRIPTION
**Pull request by salabh-aot**
- New database tables
  - works
  - eoa_teams
  - staffs
  - positions
  - roles
  - staff_work_roles
  - work_status
  - indigenous_categories
  - ~~indigenous_groups~~ indigenous_nations
  - indigenous_works
  - work_phases
 - API endpoint for staffs
 - Updated code service
 - Added `as_dict` method to `BaseModel`

### CODE REVIEW

File: src/reports_api/models/eoa_team.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from sqlalchemy import Column, Integer, String
+
+from .code_table import CodeTable
+from .db import db
+
+
+class EOATeam(db.Model, CodeTable):
+    """Model class for EOATeam."""
+
+    __tablename__ = 'eoa_teams'
```
Comment (dinesh-aot) [inline src/reports_api/models/eoa_team.py:25]: its eao. not eoa.

File: src/reports_api/resources/staff.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @cors.crossdomain(origin='*')
+    def get(position_id=None):
+        """Return all codes based on code_type."""
+        if position_id:
+            return StaffService.find_by_position_id(position_id), HTTPStatus.OK
+        else:
+            return StaffService.find_all(), HTTPStatus.OK
+
+
+# @cors_preflight('GET')
```
Comment (dinesh-aot) [inline src/reports_api/resources/staff.py:42]: You can remove this commented lines if not required.

File: src/reports_api/models/eoa_team.py (hunk 1 above)
Comment (salabh-aot) [reply src/reports_api/models/eoa_team.py]: updated

File: src/reports_api/resources/staff.py (hunk 2 above)
Comment (salabh-aot) [reply src/reports_api/resources/staff.py]: removed commented code

**Review by dinesh-aot — APPROVED** [review]
Looks good.

**Pull request merged**